//
// PROJECT:         Aspia
// FILE:            category/category_dmi_system.proto
// LICENSE:         Mozilla Public License Version 2.0
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto;

message DmiSystem
{
    enum WakeupType
    {
        WAKEUP_TYPE_UNKNOWN           = 0;
        WAKEUP_TYPE_OTHER             = 1;
        WAKEUP_TYPE_APM_TIMER         = 2;
        WAKEUP_TYPE_MODEM_RING        = 3;
        WAKEUP_TYPE_LAN_REMOTE        = 4;
        WAKEUP_TYPE_POWER_SWITCH      = 5;
        WAKEUP_TYPE_PCI_PME           = 6;
        WAKEUP_TYPE_AC_POWER_RESTORED = 7;
    }

    string manufacturer    = 1;
    string product_name    = 2;
    string version         = 3;
    string serial_number   = 4;
    string uuid            = 5;
    WakeupType wakeup_type = 6;
    string sku_number      = 7;
    string family          = 8;
}
